tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+
+            await AddSignIn_TestOnRedirectToIdentityProviderEvent(services, redirectFunc).ConfigureAwait(false);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigActions_OnRedirectToIdentityProviderEvent(bool useServiceCollectionExtension)
+        {
+            var redirectFunc = Substitute.For<Func<RedirectContext, Task>>();
+            var services = new ServiceCollection()
+                .Configure<OpenIdConnectOptions>(_oidcScheme, (options) => {
+                    options.Events ??= new OpenIdConnectEvents();
+                    options.Events.OnRedirectToIdentityProvider += redirectFunc;
+                });
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+
+            await AddSignIn_TestOnRedirectToIdentityProviderEvent(services, redirectFunc).ConfigureAwait(false);
+        }
+
+        private async Task AddSignIn_TestOnRedirectToIdentityProviderEvent(IServiceCollection services, Func<RedirectContext, Task> redirectFunc)
+        {
+            var provider = services.BuildServiceProvider();
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            var httpContext = HttpContextUtilities.CreateHttpContext();
+            var authScheme = new AuthenticationScheme(OpenIdConnectDefaults.AuthenticationScheme, OpenIdConnectDefaults.AuthenticationScheme, typeof(OpenIdConnectHandler));
+            var authProperties = new AuthenticationProperties();
+            authProperties.Items[OidcConstants.AdditionalClaims] = "additional_claims";
+            authProperties.Parameters[OpenIdConnectParameterNames.LoginHint] = "login_hint";
+            authProperties.Parameters[OpenIdConnectParameterNames.DomainHint] = "domain_hint";
+            var redirectContext = new RedirectContext(httpContext, authScheme, oidcOptions, authProperties)
+            {
+                ProtocolMessage = new OpenIdConnectMessage(),
+            };
+
+            await oidcOptions.Events.RedirectToIdentityProvider(redirectContext).ConfigureAwait(false);
+
+            // Assert properties set, events called
+            await redirectFunc.ReceivedWithAnyArgs().Invoke(Arg.Any<RedirectContext>()).ConfigureAwait(false);
+            Assert.NotNull(redirectContext.ProtocolMessage.LoginHint);
+            Assert.NotNull(redirectContext.ProtocolMessage.DomainHint);
+            Assert.NotNull(redirectContext.ProtocolMessage.Parameters[OidcConstants.AdditionalClaims]);
```

Comment:
what about checking the `redirectContext.ProtocolMessage.RequestType` and `Claims.IsAuthenticated` values as well? which is beyond something that you've intentionally set.

Reply:
Yea, I'm only testing for the output caused by the current method. That other stuff holistically should be covered by integration tests.